x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/indices_stats_collector.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import {
+  BENCHMARK_SCORE_INDEX_DEFAULT_NS,
+  FINDINGS_INDEX_DEFAULT_NS,
+  LATEST_FINDINGS_INDEX_DEFAULT_NS,
+} from '../../../../common/constants';
+import type { CspmIndicesStats, IndexStats } from './types';
+
+export const getIndicesStats = async (esClient: ElasticsearchClient): Promise<CspmIndicesStats> => {
+  const [findings, latestFindings, score] = await Promise.all([
+    getIndexStats(esClient, FINDINGS_INDEX_DEFAULT_NS),
+    getIndexStats(esClient, LATEST_FINDINGS_INDEX_DEFAULT_NS),
+    getIndexStats(esClient, BENCHMARK_SCORE_INDEX_DEFAULT_NS),
+  ]);
+  return {
+    findings,
+    latest_findings: latestFindings,
+    score,
+  };
+};
+
+const getIndexStats = async (
+  esClient: ElasticsearchClient,
+  index: string
+): Promise<IndexStats | {}> => {
+  const isIndexExists = await esClient.indices.exists({
```

Comment:
please add try-catch block, with a proper log incase of an error
better to be on the safe side here

Suggesting to enhance the returned object with `status: "error"`, wdyt?

Reply:
I added try-catch, but have little bit tied-schedule to change in this pr the scheme, will test it in the next cycle